.eslintrc.yaml

```diff
@@ -67,6 +67,43 @@ settings:
 
 rules:
 
+  #
+  # ================================================================
+  # Obsolete or redundant rules; repeal or adjust.
```

Comment:
I agree about the `ft-flow/` lines and the `react/` lines, since those are about rules from specific plugins, but I'm not yet sure about `no-undef`. Copying:

```
  # With hermes-eslint for `parser`, above, this rule can analyze Flow
  # types. It would fire on global types like $ReadOnly, for which it
  # doesn't see a declaration. We could find out how to teach the rule that
  # these types are globally defined…or just turn the rule off, because it's
  # redundant with Flow (specifically `Flow(cannot-resolve-name)`, I think):
  #   https://github.com/flow-typed/eslint-plugin-ft-flow/issues/33#issue-1274050123
  no-undef: off # https://eslint.org/docs/latest/rules/no-undef
```

`no-undef` is an out-of-the-box rule from ESLint, it's not from a plugin, so I don't yet see a section that's an exact fit. It might go under "Airbnb rule repeals / adjustments"? But:

- Is Airbnb responsible for bringing in this rule? I see `airbnb` under `extends:`, but I also see some other things there that could plausibly be asking for this rule.
- Is it important that Airbnb (etc.; see previous bullet) is bringing in this rule? Or do we want to reject it no matter where it came from, since it's redundant with Flow?

Reply:
> I also see some other things there that could plausibly be asking for this rule

To expand on this, here's the `extends:`:

```
extends:
  - airbnb
  - plugin:jest/recommended
  - plugin:react-hooks/recommended
  - ./tools/formatting.eslintrc.yaml
```

We can easily rule out `./tools/formatting.eslintrc.yaml`; it's a path to a short file we maintain. And I already knew it doesn't mention `no-undef`.

But thinking conservatively about `plugin:jest/recommended` and `plugin:react-hooks/recommended`…I'd treat "recommended for use with Jest" and "recommended for use with React Hooks" as not really helping me know what's in the tin. I guess I should read code for `eslint-plugin-jest` and `eslint-plugin-react-hooks` in `node_modules/`, or find those on GitHub and turn to the specific versions we use, with help from `yarn why`…

…ah, right, so I see that in each case, the plugin's "recommended" list only includes rules defined by the plugin itself, so in particular neither includes `no-undef`. That makes things neater, but I had to check, because nothing enforces that. (We know nothing enforces it because `airbnb` is in the same list, and brings in lots of non-plugin rules that we have existing repeals for.)

Reply:
> …ah, right, so I see that in [each](https://github.com/jest-community/eslint-plugin-jest/blob/v26.5.3/src/index.ts#L70) [case](https://github.com/facebook/react/blob/72b7462fe/packages/eslint-plugin-react-hooks/src/index.js#L16-L19), the plugin's "recommended" list only includes rules defined by the plugin itself, so in particular neither includes `no-undef`. That makes things neater, but I had to check, because nothing enforces that.

Yeah. Nothing enforces that, but OTOH my sense is that that is a pretty firm convention among ESLint plugins: A ruleset like "plugin:foo/recommended", where "foo" is an ESLint plugin, consists of a subset of the rules that the plugin "foo" itself defines.

Effectively this convention substitutes for having a concept of "this rule is enabled by default" as part of a rule's definition.

Reply:
> * Is it important that Airbnb (etc.; see previous bullet) is bringing in this rule?

Yeah, I think the "Airbnb rule repeals / adjustments" section heading may have outgrown its usefulness as part of the organization of this file.

... Hmm, in fact, I think that section never made much sense. It looks like it originates with 245ea9c4cdf8fb9459ba64dff3e71dab89b36c75 followed by 1d870ddc5e76e107e64bbaa74ef837aa60d59d39. Before that former commit, there wasn't anything like an "Airbnb repeals" section -- because that's what most of the `rules` block was about anyway.
https://github.com/zulip/zulip-mobile/blob/86a42582a/.eslintrc.yaml

So yeah, probably ideally the rules in this section would all just get moved into the appropriate thematic sections below.

Reply:
(The "Obsolete or redundant rules" section heading this comment thread was originally about is already gone in the current revision.)

Reply:
> So yeah, probably ideally the rules in this "Airbnb repeals" section would all just get moved into the appropriate thematic sections below.

(That's out of scope for this PR, though.)

Reply:
That all makes sense; thanks!